src/res_core.ml

```diff
@@ -3230,7 +3230,7 @@ and parseWhileExpression p =
 
 and parsePatternGuard p =
   match p.Parser.token with
-    | When ->
+    | When | If ->
```

Comment:
some nitpics: in general annotate the type p makes code more readable compared with `p.Parser.token`

Reply:
Thanks for the advice, will move towards the annotated style.